### PR TITLE
bipart: simplify BipartitionByIntRep

### PR DESCRIPTION
**Pull request by james-d-mitchell**

### CODE REVIEW

**Comment by james-d-mitchell** [comment]
@wilfwilson could you take a quick look at the changes made to the tests here? They relate to `IsMonogenicSemigroup` and `MaximalSubsemigroups` (1 line each). Thanks.

File: tst/standard/maximal.tst [hunk 1]
```diff
@@ -601,9 +601,8 @@ gap> MaximalSubsemigroups(U);
   <subsemigroup of 3x4 Rees 0-matrix semigroup with 5 generators> ]
 
 # maximal: MaximalSubsemigroups, for a permutation group, 1
-gap> MaximalSubsemigroups(SymmetricGroup(3));
-[ <group with 1 generator>, <group with 1 generator>, <group with 1 generator>
- , <group with 1 generator> ]
+gap> List(MaximalSubsemigroups(SymmetricGroup(3)), IsGroupAsSemigroup);
```
Comment (wilfwilson) [inline tst/standard/maximal.tst]: Fine, groups just seem to print differently now.

File: tst/standard/properties.tst [hunk 2]
```diff
@@ -941,7 +941,7 @@ gap> IsMonogenicSemigroup(S);
 true
 gap> x := MinimalSemigroupGeneratingSet(S)[1];;
 gap> S := Semigroup(x, x, x);;
-gap> HasIsMonogenicSemigroup(S) and IsMonogenicSemigroup(S);
+gap> IsMonogenicSemigroup(S);
```
Comment (wilfwilson) [inline tst/standard/properties.tst]: This isn't going to cause any problems, but I can't work out why the immediate method on `gap/attributes/properties.gi:771` no longer sets `IsMonogenicSemigroup` to be true for `S`. (Whether that immediate method is a good idea is a separate issue).

Comment (james-d-mitchell) [reply tst/standard/properties.tst]: Me neither, but as you say it is unlikely to cause any problems. Thanks!

**Pull request merged**